cilium/bpf_metadata.cc

```diff
@@ -406,38 +406,41 @@ Cilium::SocketOptionSharedPtr Config::getMetadata(Network::ConnectionSocket& soc
     // North/south L7 LB, assume the source security identity of the configured source addresses,
     // if any and policy for this identity exists.
 
-    // Pick the local source address of the same family as the incoming connection
-    const Network::Address::Ip* ip = selectIPVersion(sip->version(), source_addresses);
-
-    if (!ip) {
-      // IP family of the connection has no configured local source address
-      ENVOY_LOG(warn,
-                "cilium.bpf_metadata (north/south L7 LB): No local IP source address configured "
-                "for the family of {}",
-                pod_ip);
+    // Pick the local ingress source address of the same family as the incoming connection
+    const Network::Address::Ip* ingress_ip = selectIPVersion(sip->version(), source_addresses);
+
+    if (!ingress_ip) {
+      // IP family of the connection has no configured local ingress source address
+      ENVOY_LOG(
+          warn,
+          "cilium.bpf_metadata (north/south L7 LB): No local Ingress IP source address configured "
+          "for the family of {}",
+          sip->addressAsString());
       return nullptr;
     }
 
-    pod_ip = ip->addressAsString();
```

Comment:
IMO better not repeat `ingress_ip->addressAsString()` calls below. If you do not like setting `pod_ip` this early, how about:
```
auto& ingress_ip_str = ip->addressAsString();
```
and then later:
```
pod_ip = ingress_ip_str;
```

Reply:
thanks, good point. fixed